### PR TITLE
Feature/nginx

### PR DESCRIPTION
**Pull request by giagermain**
Nginx added for https://localhost and image uploads functionality

### CODE REVIEW

File: nginx/private.key
Comment (Eric-Butcher) [inline nginx/private.key:1]: Rename this to local private key so it is obvious that this is not the one we are using on the server.

File: evergreen/Dockerfile [hunk 1]
```diff
@@ -6,14 +6,22 @@ ENV PYTHONUNBUFFERED 1
 WORKDIR /root
 COPY pyproject.toml .
 RUN pip3 install poetry 
+
 RUN poetry config virtualenvs.create false --local
+
+# RUN apt-get update && apt-get install -y nginx
+# RUN rm /etc/nginx/conf.d/default.conf
+
```
Comment (WeirdAlex03) [inline evergreen/Dockerfile]: There's a bit of commented-out stuff like this that should just be deleted. You can look at all the changes on GitHub to see them all

File: nginx/private.key
Comment (WeirdAlex03) [reply nginx/private.key:1]: Better yet I'd take it out of the repo entirely. Anyone can just generate their own self-signed certs for testing on `localhost`

File: README.md [hunk 2]
```diff
@@ -14,6 +15,13 @@ The virtual environment can be created locally by using the `poetry install` com
 
 New package dependencies for the project can be added using the `poetry add <package-name-from-pip>` command.
 
+### Nginx for Localhost
+To set up nginx configuration, a self signed certificate is needed. To set this up, in the terminal run:
+`openssl req -x509 -newkey rsa:4096 -keyout private.key -out cert.pem -days 365 -sha256 -nodes`
```
Comment (Eric-Butcher) [inline README.md]: I would put these in triple ticks.

File: README.md [hunk 3]
```diff
@@ -14,6 +15,13 @@ The virtual environment can be created locally by using the `poetry install` com
...
+
+Only the "Country Name" field is necessary, in which you should should type "us" then press enter. All other fields may be ignored by entering "."
```
Comment (Eric-Butcher) [inline README.md]: I believe you are supposed the just press enter here, not ".". 

Alex, is this correct?

I would just tell the user to press <kbd>Enter</kbd> for all the other fields.

Comment (WeirdAlex03) [reply README.md]: Default varies by client, some will default to blank others will have placeholder data. Using <kbd>.</kbd> will leave it blank even the default isn't

File: evergreen/eg_app/views.py [hunk 4]
```diff
@@ -21,6 +21,9 @@
 import eg_app.util.validators as val
 from eg_app.models import Comments, Post
 
+# import PIL
```
Comment (Eric-Butcher) [inline evergreen/eg_app/views.py]: Why is this here

File: evergreen/eg_app/models.py [hunk 5]
```diff
@@ -7,12 +7,12 @@
...
 
-    # user = models.ForeignKey(User, on_delete=models.CASCADE)
     user = models.TextField(blank=False)
 
-    image = models.ImageField(upload_to="posts", blank=True, null=True)
+    image = models.ImageField(upload_to="image_upload", blank=True, null=True)
     caption = models.TextField(blank=True)
     timestamp = models.DateTimeField(auto_now_add=True)
+
     userLikes = models.ManyToManyField(User, blank=True)
```
Comment (Eric-Butcher) [inline evergreen/eg_app/models.py]: Why is this here?

File: evergreen/evergreen/settings.py [hunk 6]
```diff
@@ -41,6 +42,7 @@
     "django.contrib.messages",
     "django.contrib.staticfiles",
     "eg_app",
+    # "fileUpload",
```
Comment (Eric-Butcher) [inline evergreen/evergreen/settings.py]: Take this out

File: README.md [hunk 7]
```diff
@@ -14,6 +15,15 @@ The virtual environment can be created locally by using the `poetry install` com
 
 New package dependencies for the project can be added using the `poetry add <package-name-from-pip>` command.
 
+### Nginx for Localhost
+To set up nginx configuration, a self signed certificate is needed. To set this up, in the terminal run:
+```shell
+openssl req -x509 -newkey rsa:4096 -keyout private.key -out cert.pem -days 365 -sha256 -nodes
```
Comment (WeirdAlex03) [inline README.md:21]: Specify to run in `./nginx` directory

**Review by WeirdAlex03 — COMMENTED** [review]
one more note on the readme. ready to approve after that, pending @Eric-Butcher's testing locally

**Review by Eric-Butcher — APPROVED** [review]
